Base PR-Based Repro-Testing Workflow (#136)
ACCESS-NRI/access-om3-configs: CONTRIBUTING.md

### Release

Pull requests to the `release-*` branch should be made from the respective `dev-*` branch and are intended to create a new version of the configuration. These pull requests have CI quality assurance (QA) checks that ensure the model configuration is suitable for release. CI Model reproducibility checks are also conducted: a short test run of the configuration is checked for bitwise reproducibility. The success or otherwise of this check determines if a minor or major version bump is required.

It is expected that the version *will* be updated before the pull request can be merged. This in turn creates a new tag for that configuration branch. It can be confusing for users if there are a large number of versions of a configuration and it is of little benefit to them. For this reason the atomicity of updates to a released configuration should be minimised, i.e.  updates should be meaningful.

## Creation of a new ACCESS-OM3 Config

Config branches are entirely separate from the `main` history in this repository, except for a few files in `.github`. Note, you may need to be an Administrator to commit to `release-*` or `dev-*` branches directly.

> ### Config is Stored in Another Repository

Reviewer: ```suggestion
### Create a new dev branch
```

> Create a new development branch (usually prefixed by `dev-*`) by adding the config repository as a remote and checking out the config branch:
> 
> ```bash
> git remote add <config_repo> <config_repo_url>  # ex. git remote add config git@github.com/my/configs.git
> git checkout <config_repo>/<config_branch> -b <new_config_branch>  # checkout config from new remote + add to branch, ex. git checkout config/main -b dev-1deg_abc_def
> git checkout main -- .github/workflows/ci.yml
> git add .
> git commit -m "Initial commit for config branch"
> git push
> ```

Reviewer: ```suggestion
Create a new development branch with automated workflows (usually prefixed by `dev-*`) by adding the workflow file to the config branch:

```bash
git checkout dev-<new_branch_name>
git checkout main -- .github/workflows/ci.yml
git add .
git commit -m "Initial commit for config branch"
git push
```
```

### Release

Pull requests to the `release-*` branch should be made from the respective `dev-*` branch and are intended to create a new version of the configuration. These pull requests have CI quality assurance (QA) checks that ensure the model configuration is suitable for release. CI Model reproducibility checks are also conducted: a short test run of the configuration is checked for bitwise reproducibility. The success or otherwise of this check determines if a minor or major version bump is required.

It is expected that the version *will* be updated before the pull request can be merged. This in turn creates a new tag for that configuration branch. It can be confusing for users if there are a large number of versions of a configuration and it is of little benefit to them. For this reason the atomicity of updates to a released configuration should be minimised, i.e.  updates should be meaningful.

## Creation of a new ACCESS-OM3 Config

Config branches are entirely separate from the `main` history in this repository, except for a few files in `.github`. Note, you may need to be an Administrator to commit to `release-*` or `dev-*` branches directly.

> ### Brand new configuration
> 
> If you are creating a brand new configuration, and don't have the config stored in another repository, just checkout a `dev-*` branch from `main` and delete everything except `.github/workflows/ci.yml`, then add your config.

Reviewer: ```suggestion

```

This is covered in the Readme

> ### Config is Stored in Another Repository

Reviewer: This section can be generalised a little - i..e. Create a new dev branch

Author: Done in https://github.com/ACCESS-NRI/access-om3-configs/pull/136/commits/03d0d09bc899b2dbcb639b21775efba46a833640
